test/sslapitest.c

```diff
@@ -6684,7 +6684,60 @@ static int test_ticket_callbacks(int tst)
  *                  sends CertificateRequest, client reads and ignores it
  * Test 6: TLSv1.3, server continues to read/write after client shutdown, client
  *                  doesn't read it
+ * Test 7: TLSv1.3, client does not shutdown properly,
+ *                  server does not set SSL_OP_IGNORE_UNEXPECTED_EOF,
+ *                  server should get SSL_ERROR_SSL
+ * Test 8: TLSv1.3, client does not shutdown properly,
+ *                  server sets SSL_OP_IGNORE_UNEXPECTED_EOF,
+ *                  server should get SSL_ERROR_ZERO_RETURN
  */
+
+static int test_incorrect_shutdown(int tst)
```

Comment:
Since the codepath is entirely different I would prefer this to be an independent top-level test instead of a sub-flow of `test_shutdown`

Reply:
Done. Thanks!

Reply:
Done. Thanks!